### PR TITLE
refactor: renamed UI to ColorConfig

### PR DESCRIPTION
**Pull request by NicholasLYang**
### Description

Since we now have an explicit `ui` config that refers to whether we display runs via stream, tui, or web, we should rename the existing `UI` struct to be more explicitly about color configuration (since that's the only thing it does).

### CODE REVIEW

File: crates/turborepo-auth/src/auth/logout.rs [hunk 1]
```diff
@@ -171,7 +171,7 @@ mod tests {
             .expect("could not create file");
 
         let logout_options = LogoutOptions {
-            ui: UI::new(false),
+            ui: ColorConfig::new(false),
```
Comment (dimitropoulos) [inline crates/turborepo-auth/src/auth/logout.rs:174]: was it intentional to also keep this named `ui`?

File: crates/turborepo-auth/src/auth/mod.rs [hunk 2]
```diff
@@ -46,7 +46,7 @@ impl<'a, T: Client + TokenClient + CacheClient> LoginOptions<'a, T> {
 
 /// Options for logging out.
 pub struct LogoutOptions<T> {
-    pub ui: UI,
+    pub ui: ColorConfig,
```
Comment (dimitropoulos) [inline crates/turborepo-auth/src/auth/mod.rs:49]: was it intentional to also keep this named `ui`?  is this exposed to users?

File: crates/turborepo-auth/src/auth/mod.rs [hunk 3]
```diff
@@ -27,13 +27,13 @@ pub struct LoginOptions<'a, T: Client + TokenClient + CacheClient> {
 }
 impl<'a, T: Client + TokenClient + CacheClient> LoginOptions<'a, T> {
     pub fn new(
-        ui: &'a UI,
+        ui: &'a ColorConfig,
```
Comment (dimitropoulos) [inline crates/turborepo-auth/src/auth/mod.rs:30]:
```suggestion
        color_config: &'a ColorConfig,
```

File: crates/turborepo-auth/src/ui/messages.rs [hunk 4]
```diff
@@ -1,6 +1,6 @@
-use turborepo_ui::{BOLD, CYAN, UI};
+use turborepo_ui::{ColorConfig, BOLD, CYAN};
 
-pub fn print_cli_authorized(user: &str, ui: &UI) {
+pub fn print_cli_authorized(user: &str, ui: &ColorConfig) {
```
Comment (dimitropoulos) [inline crates/turborepo-auth/src/ui/messages.rs:3]:
```suggestion
pub fn print_cli_authorized(user: &str, color_config: &ColorConfig) {
```

File: crates/turborepo-lib/src/commands/ls.rs [hunk 5]
```diff
@@ -25,13 +25,13 @@ pub enum Error {
...
 struct RepositoryDetails<'a> {
-    ui: UI,
+    ui: ColorConfig,
     package_manager: &'a PackageManager,
     packages: Vec<(&'a PackageName, &'a AnchoredSystemPath)>,
 }
 
 struct PackageDetails<'a> {
-    ui: UI,
+    ui: ColorConfig,
```
Comment (dimitropoulos) [inline crates/turborepo-lib/src/commands/ls.rs:34]: just verifying that both of these were intended not to be renamed

File: crates/turborepo-lib/src/commands/mod.rs [hunk 6]
```diff
@@ -43,11 +43,11 @@ impl CommandBase {
         args: Args,
         repo_root: AbsoluteSystemPathBuf,
         version: &'static str,
-        ui: UI,
+        ui: ColorConfig,
```
Comment (dimitropoulos) [inline crates/turborepo-lib/src/commands/mod.rs:46]:
```suggestion
        color_config: ColorConfig,
```

File: crates/turborepo-lib/src/run/builder.rs [hunk 7]
```diff
@@ -54,7 +54,7 @@ pub struct RunBuilder {
     opts: Opts,
     api_auth: Option<APIAuth>,
     repo_root: AbsoluteSystemPathBuf,
-    ui: UI,
+    ui: ColorConfig,
```
Comment (dimitropoulos) [inline crates/turborepo-lib/src/run/builder.rs:57]: just verifying this wasn't intended to be renamed

**Review by dimitropoulos — APPROVED** [review]
didn't see anything major - just a few double-checks and one or two chances to clean up constructors that don't need argument renaming

**Review by chris-olszewski — APPROVED** [review]

**Pull request merged**